provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1243,11 +1284,17 @@ func generateSchema() schema.PackageSpec {
 						"See https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/.",
 					Properties: map[string]schema.PropertySpec{
 						"value": {
-							TypeSpec:    schema.TypeSpec{Type: "string"},
+							TypeSpec: schema.TypeSpec{
+								Type:  "string",
+								Plain: true,
+							},
 							Description: "The value of the taint.",
 						},
 						"effect": {
-							TypeSpec:    schema.TypeSpec{Type: "string"}, // TODO strict string enum: "NoSchedule" | "NoExecute" | "PreferNoSchedule".
+							TypeSpec: schema.TypeSpec{
+								Type:  "string", // TODO strict string enum: "NoSchedule" | "NoExecute" | "PreferNoSchedule".
+								Plain: true,
+							},
 							Description: "The effect of the taint.",
 						},
```

Comment:
These are plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/nodegroup.ts#L31-L40